### PR TITLE
Add OpenRouter, with Mixtral demo

### PR DESCRIPTION
**Pull request by alexanderatallah**
# Description

Adds support for OpenRouter: openrouter.ai. Also adds a demo Collab file using Mixtral 8x7b: https://openrouter.ai/models/mistralai/mixtral-8x7b-instruct

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/bridge/langchain.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "Cohere",
     "BaseChatModel",
     "ChatAnyscale",
+    "ChatOpenRouter",
```
Comment (logan-markewich) [inline llama_index/bridge/langchain.py:64]: Is this actually needed?

File: docs/examples/llm/openrouter.ipynb
Comment (logan-markewich) [inline docs/examples/llm/openrouter.ipynb:1]: Can we add some description to this? What is open router? How do I get an API key?

File: llama_index/llms/openrouter.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+)
+from llama_index.llms.generic_utils import get_from_param_or_env
+from llama_index.llms.openai import OpenAI
+from llama_index.llms.types import LLMMetadata
+
+DEFAULT_API_BASE = "https://openrouter.ai/api/v1"
+DEFAULT_MODEL = "gryphe/mythomax-l2-13b"
+
+
+class OpenRouter(OpenAI):
```
Comment (hatianzhang) [inline llama_index/llms/openrouter.py:17]: should OpenRouter be a subclass of OpenAI?
if it is based on other LLMs, could we do something standalone like openai_like.py

File: llama_index/bridge/langchain.py (hunk 1 above)
Comment (logan-markewich) [reply llama_index/bridge/langchain.py:64]: Removed

File: llama_index/llms/openrouter.py (hunk 2 above)
Comment (logan-markewich) [reply llama_index/llms/openrouter.py]: yea, just inherited openai like (needs the extra hook for api key)

**Pull request merged**